cost.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package celvet
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/google/cel-go/cel"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation"
+	structuralschema "k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	schemacel "k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel"
+)
+
+type costError struct {
+	Program cel.Program
+	Name    string
+	Cost    uint64
+}
+
+func (c *costError) Error() string {
+	exceedFactor := float64(c.Cost) / float64(validation.StaticEstimatedCostLimit)
+	return fmt.Sprintf("expression at %q exceeded budget by factor of %.1fx", c.Name, exceedFactor)
```

Comment:
The message seems confusing to me

Reply:
Would `expression at %q too expensive to execute by factor of %.1fx` be better, maybe?

Reply:
I was thinking something like `c.Name: the cost of expression is c.Cost which exceeds the cost limit of validation.StaticEstimatedCostLimit`

Reply:
> I'm not quite sure having large numbers like that makes the error message more readable.

I agree it doesn't make the error message more readable, but need to balance that with the fact that this is our diagnostic tool. If we're going to to have a tool that can tell us the exact costs of things (which I think is something we want), it should be this tool. I'm flexible on how the exact costs are surfaced (I'd be okay with having to set a flag to see them), but they should be accessible.